test_runner/batch_others/test_remote_storage.py

```diff
@@ -54,13 +55,21 @@ def test_remote_storage_backup_and_restore(zenith_env_builder: ZenithEnvBuilder,
                 CREATE TABLE t1(id int primary key, secret text);
                 INSERT INTO t1 VALUES ({data_id}, '{data_secret}');
             ''')
+            cur.execute("SELECT pg_current_wal_flush_lsn()")
+            current_lsn = int(cur.fetchone()[0].split('/')[1], base=16)
+
+    # wait until pageserver receives that data
+    wait_for_last_record_lsn(client, UUID(tenant_id), UUID(timeline_id), current_lsn)
 
     # run checkpoint manually to be sure that data landed in remote storage
     with closing(env.pageserver.connect()) as psconn:
         with psconn.cursor() as pscur:
             pscur.execute(f"do_gc {tenant_id} {timeline_id}")
```

Comment:
> do_gc

Let's replace it with `checkpoint` here and maybe in the replication test too? Seems like a one word change that can save us a lot of trouble.

Totally fine with a separate follow-up task for this.

Reply:
For some reason I had to increase the margin percent in checks to 0.03 after that change, previously with gc there was no lag at all

Reply:
That's odd, but thanks for looking into that.

If I've got it right, Lsm PR might change the way we approach the whole GC approach, so the timing to change this is great anyway.